src/Xamarin.Android.Build.Tasks/Resources/LayoutBinding.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+
+using Android.App;
+using Android.Views;
+
+namespace Xamarin.Android.Design
+{
+	abstract class LayoutBinding
+	{
+		public delegate Java.Lang.Object OnLayoutItemNotFoundHandler (int resourceId, Type expectedViewType);
+
+		Activity boundActivity;
+		View boundView;
+		OnLayoutItemNotFoundHandler onLayoutItemNotFound;
+
+		public abstract int ResourceLayoutID { get; }
+
+		protected LayoutBinding (Activity activity, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundActivity = activity ?? throw new ArgumentNullException (nameof (activity));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected LayoutBinding (View view, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundView = view ?? throw new ArgumentNullException (nameof (view));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected T FindView <T> (int resourceId, ref T cachedField) where T: View
+		{
+			if (cachedField != null)
+				return cachedField;
+
+			T ret;
+			if (boundActivity != null)
+				ret = boundActivity.FindViewById <T> (resourceId);
+			else
+				ret = boundView.FindViewById <T> (resourceId);
+
+			if (ret == null && onLayoutItemNotFound != null)
+				ret = (T)onLayoutItemNotFound (resourceId, typeof (T));
+
+			if (ret == null)
+				throw new global::System.InvalidOperationException ($"View not found (Layout ID: {ResourceLayoutID}; Resource ID: {resourceId})");
+
+			cachedField = ret;
+			return ret;
+		}
+#if __ANDROID_11__
+		protected global::Android.App.Fragment FindFragment (int resourceId, ref global::Android.App.Fragment cachedField)
+		{
+			if (cachedField != null)
+				return cachedField;
+
+			if (boundActivity == null)
+				throw new InvalidOperationException ("Finding fragments is supported only for Activity instances");
```

Comment:
I think this would be supportable if we updated the `LayoutBinding(View)` constructor to set `boundActivity`:

```csharp
boundView = view ?? throw new ArgumentNullException (nameof (view))
boundActivity = boundView.Context as Activity;
```